### PR TITLE
[DROOLS-3368] Extend scenario runner to support DMN runtime

### PR DESCRIPTION
**Pull request by danielezonca**
NOTE: it cannot be tested via UI yet, only code + tests

@kkufova @Rikkola @gitgabrio 

PR list
https://github.com/kiegroup/drools-wb/pull/1027
https://github.com/kiegroup/drools/pull/2194
https://github.com/kiegroup/droolsjbpm-knowledge/pull/353

### CODE REVIEW

**Review by Rikkola — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/AbstractDMNModelCommand.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    .ofNullable(dmnRuntime.getModel(namespace, modelName))
+                    .orElseThrow(() -> new IllegalStateException("Cannot find a DMN model with namespace=" + namespace + " and modelName=" + modelName));
+        }
+        else if(resourcePath != null) {
+            return dmnRuntime.getModels().stream()
+                    .filter(model -> Objects.equals(resourcePath, model.getResource().getSourcePath()))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalStateException("Cannot find a DMN model with resource=" + resourcePath));
+        }
+        throw new IllegalStateException("This should not happen");
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/AbstractDMNModelCommand.java]: I see this was just moved here, however please change the error message here. This doesn't say anything. E.g. "Cannot retrieve the model! Resource path or model name is not set properly."

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        ks = KieServices.Factory.get();
+        releaseId = ks.newReleaseId("org.kie", "dmn-test-" + UUID.randomUUID(), "1.0");
+        resource = ks.getResources().newClassPathResource("org/kie/dmn/core/typecheck_in_DT.dmn", GetDMNModelCommandTest.class);
+        kieContainer = KieHelper.getKieContainer(releaseId, resource);
+    }
+
+    @Before
+    public void init() {
+        registryContext = new ContextImpl();
+
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java]: Nitpicking (sorry:)): unnecessary empty line.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    @Test
+    public void execute() {
+        String namespace = "http://www.trisotech.com/definitions/_99ccd4df-41ac-43c3-a563-d58f43149829";
+        String modelName = "typecheck in DT";
+        GetDMNModelCommand getDMNModelCommand = new GetDMNModelCommand(namespace, modelName);
+
+        try {
+            getDMNModelCommand.execute(registryContext);
+            fail();
+        } catch (IllegalStateException ignored) {
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java]: Is this exception a correct state? This looks incorrect. This kind of catches in tests could hide potential bugs in the future.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Test
+    public void executeWithResource() {
+        GetDMNModelCommand getDMNModelCommand = new GetDMNModelCommand(resource.getSourcePath());
+
+        try {
+            getDMNModelCommand.execute(registryContext);
+            fail();
+        } catch (IllegalStateException ignored) {
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java]: Same as previous comment.

**Review by baldimir — CHANGES_REQUESTED** [review]

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/AbstractDMNModelCommand.java (hunk 1 above)
Comment (danielezonca) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/AbstractDMNModelCommand.java]: Changed to "Both resourcePath and namespace+modelName are not set, this should not happen". It is not possible that both are `null`, it is just to have an error if in the future the logic will change without update this method

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java (hunk 2 above)
Comment (danielezonca) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java]: Done

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java (hunk 3 above)
Comment (danielezonca) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java]: Migrated all these `catch(IllegalStateException ignored)` to Assertj with a constraint on the actual message

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java (hunk 4 above)
Comment (danielezonca) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/fluent/GetDMNModelCommandTest.java]: Migrated all these `catch(IllegalStateException ignored)` to Assertj with a constraint on the actual message

**Review by baldimir — APPROVED** [review]
Thanks, approving.

**Review by kkufova — APPROVED** [review]
Looks good to me.

:+1:

**Pull request merged**